File: Sources/Gamified/Services/GlobalStorageService.swift
//
//  GlobalStorageService.swift
//  
//
//  Created by Nathan Fallet on 15/03/2023.
//

import Foundation

public protocol GlobalStorageService {
    
    /// Get a value
    /// - Parameter key: The value to get
    /// - Returns: The value
    func getValue(_ key: String) -> Int
    
    /// Set a value
    /// - Parameters:
    ///   - key: The value to set
    ///   - value: The value to set
    func setValue(_ key: String, value: Int)
    
}

File: Gamified/src/main/java/me/nathanfallet/gamified/services/GlobalStorageService.kt
package me.nathanfallet.gamified.services

interface GlobalStorageService {

    /**
     * Get a value
     * @param key The value to get
     * @return The value
     */
    fun getValue(key: String): Long

    /**
     * Set a value
     * @param key The value to set
     * @param value The value to set
     */
    fun setValue(key: String, value: Long)

}